transactions/demo-tx-village/tests/haskell_integration_test.rs

```diff
@@ -0,0 +1,295 @@
+mod common;
+
+use lbf_demo_plutus_api::demo::request::{ClaimRequest, LockRequest, Request};
+use lbf_demo_plutus_api::demo::response::Response;
+use num_bigint::BigInt;
+use plutus_ledger_api::json::Json;
+use plutus_ledger_api::v1::transaction::POSIXTime;
+use serial_test::serial;
+use std::io::Write;
+use std::process::Command;
+use std::time::SystemTime;
+use tx_bakery::wallet::Wallet;
+
+/// Calls the `demo-haskell` CLI to produce a `Response` suitable for creating a lock tx
+fn demo_haskell_lock_response(config_path: &str, request: Request<LockRequest>) -> Response<()> {
+    let mut temp_file = tempfile::NamedTempFile::new().unwrap();
+
+    let json_string_request = request.to_json_string();
+
+    let _ = temp_file.write(json_string_request.as_bytes()).unwrap();
+
+    let output = Command::new("demo-haskell")
+        .env("DEMO_CONFIG", config_path)
+        .arg("build-tx-info")
+        .arg("lock")
+        .arg("--request")
+        .arg(temp_file.path().to_str().unwrap())
+        .output()
+        .unwrap();
```

Comment:
Now it will make sense when I tell you: I want this to be a CLI interface in the true sense of the word, so ALL our PLA TxInfo producing backends follows the same thing if that makes sense.

Then here you'd just provide a name of the CLI and that's it.

A single test orchestration for all backends.